Add test suite and initial support for PyPI publishing via Travis CI

hectcastro opened the pull request:
This changeset adds a basic `unittest` suite for this project, along with instructions for executing it locally and inside of a Docker container. In addition, support for Travis CI was added to execute the test suite after linting.

PyPI deployment has not been tested, but it will only be triggered on Git tags.

Attempts to resolve #4 and #5.

notthatbreezy commented on setup.py:
```diff
@@ -1,12 +1,32 @@
+          'DESCRIPTION.rst'), encoding='utf-8') as f:
+    long_description = f.read()
+
+tests_require = ['moto >=0.4.1']
```
moto looks really cool

sharph commented:
I'm a fan of 80 column lines for the linter, and that's what I was aiming for when I wrote it (mostly to make flake8 happy,) but the JEST standard is 100 columns. @ddohler & @notthatbreezy do you have thoughts on this?

notthatbreezy commented:
I like 100 columns personally

sharph commented:
LGTM, aside from the line length issue

hectcastro commented:
Added `flake8` configuration file with settings for `100` column length lines. Also reverted the changes that were made to accommodate the default column width setting.

notthatbreezy commented:
+1
